themes/src/main/resources/theme/keycloak.v2/login/resources/js/password-policy.js

```diff
@@ -11,15 +11,15 @@ const policies = {
   },
   upperCase: (policy, value) => {
     if (
-      value.split("").filter((char) => char !== char.toUpperCase()).length >
+      value.split("").filter((char) => char === char.toUpperCase() && char !== char.toLowerCase()).length <
```

Comment:
If `char` is the same as `char.toUpperCase()` then why would we still need to check the lowercase? Same for the other condition, but inversed.

Reply:
I see, perhaps then a regular expression would be better suited for this? For example:

```js
value.match(/[A-Z]/g)?.length ?? 0; // This should give a count of all uppercase alpha characters.
```

Reply:
But afaik characters like ä, ö and ü are not included in `[a-z]`.

Reply:
Fair enough, let's consider this resolved then.